repo/fsrepo/migrations/versions_test.go

```diff
@@ -0,0 +1,76 @@
+package migrations
+
+import (
+	"context"
+	"os/exec"
+	"testing"
+
+	"github.com/coreos/go-semver/semver"
+)
+
+const testDist = "go-ipfs"
+
+func TestDistVersions(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	fetcher := NewHttpFetcher()
+	ts := createTestServer()
+	defer ts.Close()
+	err := fetcher.SetGateway(ts.URL)
+	if err != nil {
+		panic(err)
+	}
+
+	vers, err := DistVersions(ctx, fetcher, testDist, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(vers) == 0 {
+		t.Fatal("no versions of", testDist)
+	}
+	t.Log("There are", len(vers), "versions of", testDist)
+	t.Log("Latest 5 are:", vers[:5])
+}
+
+func TestLatestDistVersion(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	fetcher := NewHttpFetcher()
+	//ts := createTestServer()
+	//defer ts.Close()
+	//fetcher.SetGateway(ts.URL)
+
+	latest, err := LatestDistVersion(ctx, fetcher, testDist, false)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(latest) < 6 {
+		t.Fatal("latest version string too short", latest)
+	}
+	_, err = semver.NewVersion(latest[1:])
+	if err != nil {
+		t.Fatal("latest version has invalid format:", latest)
+	}
+	t.Log("Latest version of", testDist, "is", latest)
+}
+
+func TestIpfsRepoVersion(t *testing.T) {
+	_, err := exec.LookPath("ipfs")
```

Comment:
again, bad idea to use exec but also not sure why this function (+ test are needed)

Reply:
moving back to ipfs-update

Reply:
Done.